in-progress/8757-fees/design.md

```diff
@@ -0,0 +1,270 @@
+```
+
+A proposer can adjust the `fee_asset_price_modifier` stored in the rollup contract by up to `MAXIMUM_FEE_ASSET_PRICE_PERCENT_CHANGE_PER_BLOCK` each slot.
+
+They do so by updating the `fee_asset_price_modifier_delta` field, which is used to update the `fee_asset_price_modifier` stored in the rollup contract.
+
+That is,
+
+```math
+\text{new fee asset price modifier} := \text{old fee asset price modifier} + \text{fee asset price modifier delta}
+```
+
+The new `fee_asset_price_modifier_delta` is capped at (+/-) `MAXIMUM_FEE_ASSET_PRICE_PERCENT_CHANGE_PER_BLOCK` * `FEE_ASSET_PRICE_UPDATE_FRACTION` / 100.
+
+### `base_fee_wei_per_mana_congestion_multiplier`
+
+First we compute the excess mana in the current block by considering the parent mana spent and excess mana.
+
+```math
+\text{excess mana} = \begin{cases} 
+0 & \text{if } \text{parent.excess} + \text{parent.spent} < \text{TARGET MANA PER BLOCK} \\
+\text{parent.excess} + \text{parent.spent} - \text{TARGET MANA PER BLOCK} & \text{otherwise}
+\end{cases}
+```
+
+```math
+\begin{aligned}
+\text{base fee wei per mana congestion multiplier} &= \text{MINIMUM CONGESTION MULTIPLIER} * \text{exp}\left(\frac{\text{excess mana}}{\text{CONGESTION MULTIPLIER UPDATE FRACTION}}\right)
+\end{aligned}
+```
+
+## Deriving the base fee
+
+When a proposer is building a block, it calculates:
+```math
+\begin{aligned}
+\text{block cost in wei} &= \text{execution cost in wei} + \text{data cost in wei} \\
+\text{block cost in wei/mana} &= \frac{\text{block cost in wei}}{\text{TARGET MANA PER BLOCK}} + \text{proving cost wei per mana} \\
+\text{base fee in wei/mana} &= \text{block cost in wei/mana} * \text{base fee wei per mana congestion multiplier} \\
+\text{base fee in fee asset/mana} &= \text{base fee in wei/mana} * \text{fee asset per wei}
+\end{aligned}
+```
+
+This final value is the `base_fee_asset_per_mana` field in the L2 block header.
+
+## The cost of a transaction in the fee asset
+
+The amount of mana a transaction consumes is:
+
+```math
+\text{mana}_{tx} = \text{OVERHEAD MANA PER TX} + \sum_{op \in tx} \text{mana per operation}(op)
+```
+
+Therefore, the cost of a transaction's proposal, data publication, and verification at time $i$ is, denominated in the fee asset:
+
+```math
+\text{fee asset}_{tx} = \text{mana}_{tx} * \text{base fee asset per mana}
+```
+
+## Distribution of the fees
+
+When a transaction is included in a block in the pending chain, the L2 balance of the fee asset for the `fee_payer` of the transaction is reduced by the transactions's fee asset cost.
+
+When an epoch is proven, the fees for the epoch are paid out to each proposer for the block(s) they proposed.
+
+A percentage of each block's fee asset fees are paid to the prover of the epoch. The percentage is determined based on the quote submitted by the prover for the epoch.
```

Comment:
Just checking this includes block rewards right? 

Otherwise the network could not subsidize proving, users would have to pay the full proving cost. 

We could have: 
1) MAXIMUM_EPOCH_PROOF_QUOTE_FEE = % (block fees + block rewards) or
2) provers always get `proving wei per mana  * total_mana_per_block`  + MAXIMUM_EPOCH_PROOF_QUOTE_FEE% * total_block_reward.

Reply:
Block rewards were intentionally not discussed because they are not part of the fee mechanism.

What is implemented today on master is that the prover gets paid in Fee Asset:
```math
\text{proofClaim.basisPointFee} \times \sum_{\text{block} \in \text{epoch}} \text{Block Reward} + \text{block.fees} 
```